pkg/infrastructure/nutanix/clusterapi/clusterapi.go

```diff
@@ -77,45 +77,49 @@ func (p Provider) PreProvision(ctx context.Context, in infracapi.PreProvisionInp
 	}
 	logrus.Infof("created the category value %q with name %q", *respv.Value, *respv.Name)
 
-	// upload the rhcos image.
-	imgName := nutanixtypes.RHCOSImageName(in.InfraID)
-	imgURI := in.RhcosImage.ControlPlane
-	imgReq := &nutanixclientv3.ImageIntentInput{}
-	imgSpec := &nutanixclientv3.Image{
-		Name:        &imgName,
-		Description: ptr.To("Created By OpenShift Installer"),
-		Resources: &nutanixclientv3.ImageResources{
-			ImageType: ptr.To("DISK_IMAGE"),
-			SourceURI: &imgURI,
-		},
-	}
-	imgReq.Spec = imgSpec
-	imgMeta := &nutanixclientv3.Metadata{
-		Kind:       ptr.To("image"),
-		Categories: map[string]string{categoryKey: nutanixtypes.CategoryValueOwned},
-	}
-	imgReq.Metadata = imgMeta
-	respi, err := nutanixCl.V3.CreateImage(ctx, imgReq)
-	if err != nil {
-		return fmt.Errorf("failed to create the rhcos image %q: %w", imgName, err)
-	}
-	imgUUID := *respi.Metadata.UUID
-	logrus.Infof("creating the rhcos image %s (uuid: %s).", imgName, imgUUID)
+	if in.InstallConfig.Config.Nutanix.PreloadedOSImageName != "" {
+		logrus.Infof("use the existing rhcos image %q in PC, instead of create a new image object for the cluster.", in.InstallConfig.Config.Nutanix.PreloadedOSImageName)
```

Comment:
Isn't this enough?
```suggestion
		logrus.Infof("Using existing rhcos image %q in PC", in.InstallConfig.Config.Nutanix.PreloadedOSImageName)
		return nil
```
Also consider return early so you can drop the following `else` and decrese the indent level of the next block of code.

Reply:
changed the log message as suggested.